sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -405,13 +406,14 @@ case class WrapOption(child: Expression, optType: DataType)
  * A place holder for the loop variable used in [[MapObjects]].  This should never be constructed
  * manually, but will instead be passed into the provided lambda function.
  */
-case class LambdaVariable(value: String, isNull: String, dataType: DataType) extends LeafExpression
+case class LambdaVariable(value: String, isNull: String, dataType: DataType,
+    valueNullable: Boolean = true) extends LeafExpression
```

Comment:
I meant that we could use the parameter name `nullable` like:

```scala
case class LambdaVariable(value: String, isNull: String, dataType: DataType,
    nullable: Boolean = true) extends LeafExpression
```

and remove `override def nullable: Boolean = valueNullable`.